packages/emitter-framework/src/core/context/tsp-context.ts

```diff
@@ -0,0 +1,24 @@
+import { ComponentContext, createNamedContext, useContext } from "@alloy-js/core";
+import { Program } from "@typespec/compiler";
+import { unsafe_$ } from "@typespec/compiler/experimental";
+import { Typekit } from "@typespec/compiler/experimental/typekit";
+
+export type TspContext = {
+  program: Program;
+  $: Typekit;
+};
+
+export const TspContext: ComponentContext<{ program: Program; $?: Typekit }> = createNamedContext<{
+  program: Program;
+  $?: Typekit;
+}>("TspContext");
+
+export function useTsp() {
+  const context = useContext(TspContext)!;
+
```

Comment:
I would maybe throw an error here saying that the context is not set and explain it should be wrapped in the `Output` component